File: Client/astroPaper_client.py
from __future__ import print_function

import grpc
import re

import astropaperservice_pb2
import astropaperservice_pb2_grpc

def clearReply(reply_string):
    reply_string = str(reply_string)
    reply_string = reply_string[8:]
    reply_string = reply_string[:-2]
    return reply_string

def run():
    print("Client is running")
    channel = grpc.insecure_channel('localhost:50050')
    stub = astropaperservice_pb2_grpc.AstroPaperServiceStub(channel)
    print("GET request")
    wallpaper = stub.GetNewWallpaper(astropaperservice_pb2.WallpaperRequest(quantity=1))
    wallpaper = clearReply(wallpaper)
    print("Cleared reply: " + wallpaper )
    #print("Wallpaper : " + str(wallpaper))
    setup_reply = stub.SetupWallpaper(astropaperservice_pb2.SetupRequest(wallpaper=str(wallpaper)))
if __name__ == '__main__':
    run()

File: Client/astropaper_client.js
var messages = require('./apservice_pb');
var services = require('./apservice_grpc_pb');
var EventEmitter = require('events').EventEmitter;
var grpc = require('grpc');

function main() {
  console.log("Client is running");
  var client = new services.AstropaperClient('localhost:50050',
                                       grpc.credentials.createInsecure());
  var request = new messages.WallpaperRequest();
  
  var quantity;
  if (process.argv.length >= 3) {
    quantity = process.argv[2];
  } else {
    quantity = 1;
  }

  request.setQuantity(quantity);
  client.getNewWallpaper(request, function(err, response) {
    console.log('Wallpaper:', response.array[0]);
    var wallpaper = response.array[0];
    // chained get + setup
    var setup = new messages.SetupRequest();
    setup.setWallpaper(wallpaper);
    client.setupWallpaper(setup, function(err, response) {
      console.log('', wallpaper);
    });
  });
}

main();